### PR TITLE
refactor(api): DEPRECATED_GitClone migration

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/action/action.go [hunk 1]
```diff
@@ -465,3 +465,87 @@ func isRequirementsValid(requirements sdk.RequirementList) error {
 	}
 	return nil
 }
+
+type PipelineUsingAction struct {
```
Comment (sguiheux) [inline engine/api/action/action.go]: doc

File: engine/api/action/action.go [hunk 2]
```diff
@@ -465,3 +465,87 @@ func isRequirementsValid(requirements sdk.RequirementList) error {
...
+	AppName          string `json:"application_name"`
+	ProjName         string `json:"project_name"`
+	ProjKey          string `json:"key"`
+	StageID          int64  `json:"stage_id"`
+	WorkflowName     string `json:"workflow_name"`
+	WorkflowNodeName string `json:"workflow_node_name"`
+	WorkflowNodeID   int64  `json:"workflow_node_id"`
+}
+
+func GetPipelineUsingAction(db gorp.SqlExecutor, name string) ([]PipelineUsingAction, error) {
```
Comment (sguiheux) [inline engine/api/action/action.go]: doc

File: engine/api/bootstrap/bootstrap.go [hunk 3]
```diff
@@ -48,3 +55,176 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
 
 	return nil
 }
+
+const DEPRECATEDGitClone = "DEPRECATED_GitClone"
+
+// MigrateActionDEPRECATEDGitClone is temporary code
+func MigrateActionDEPRECATEDGitClone(DBFunc func() *gorp.DbMap, store cache.Store) error {
+	pipelines, err := action.GetPipelineUsingAction(DBFunc(), DEPRECATEDGitClone)
```
Comment (sguiheux) [inline engine/api/bootstrap/bootstrap.go]: Just select ID

File: engine/api/bootstrap/bootstrap.go [hunk 4]
```diff
@@ -48,3 +55,176 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
...
+
+	for _, p := range pipelines {
+		log.Info("MigrateActionDEPRECATEDGitClone> Migrate %s/%s", p.ProjKey, p.PipName)
+
+		tx, err := DBFunc().Begin()
+		if err != nil {
+			return sdk.WrapError(err, "MigrateActionDEPRECATEDGitClone> Cannot start transaction")
+		}
+
+		if err := MigrateActionDEPRECATEDGitClonePipeline(tx, store, p); err != nil {
```
Comment (sguiheux) [inline engine/api/bootstrap/bootstrap.go]: select + lock pip

File: engine/api/bootstrap/bootstrap.go [hunk 5]
```diff
@@ -48,3 +55,176 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
...
+
+	//Load the application
+	log.Debug("load application %s", appName)
+	app, err := application.LoadByName(db, store, pkey, appName, nil, application.LoadOptions.WithKeys)
+	if err != nil {
+		return err
+	}
+	appKeys := app.SSHKeys()
+
+	//Check all the steps of the job
```
Comment (sguiheux) [inline engine/api/bootstrap/bootstrap.go]: env keys ?

File: engine/api/bootstrap/bootstrap.go [hunk 6]
```diff
@@ -48,3 +55,176 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
...
+
+			//Keep the old parameters
+			sdk.ParameterFind(&newGitClone.Parameters, "url").Value = url.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "directory").Value = directory.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "branch").Value = branch.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "commit").Value = commit.Value
+			sdk.ParameterFind(&newGitClone.Parameters, "user").Value = ""
+			sdk.ParameterFind(&newGitClone.Parameters, "password").Value = ""
+
+			//If there is an application key or a project key, use it
```
Comment (sguiheux) [inline engine/api/bootstrap/bootstrap.go]: if url.value == git.http_url do nothing?

File: engine/api/bootstrap/bootstrap.go (hunk 3 above)
Comment (fsamin) [reply engine/api/bootstrap/bootstrap.go]: nope

File: sdk/application.go [hunk 7]
```diff
@@ -41,6 +41,28 @@ type Application struct {
 	Usage              *Usage                `json:"usage,omitempty" db:"-" cli:"-"`
 }
 
+// SSHKeys returns the slide of ssh key for an application
```
Comment (bnjjj) [inline sdk/application.go]: s/slide/slice

File: sdk/application.go [hunk 8]
```diff
@@ -41,6 +41,28 @@ type Application struct {
...
+	keys := []ApplicationKey{}
+	for _, k := range a.Keys {
+		if k.Type == KeyTypeSSH {
+			keys = append(keys, k)
+		}
+	}
+	return keys
+}
+
+// PGPKeys returns the slide of pgp key for an application
```
Comment (bnjjj) [inline sdk/application.go]: s/slide/slice

File: sdk/project.go [hunk 9]
```diff
@@ -33,6 +33,28 @@ type Project struct {
 	Platforms         []ProjectPlatform  `json:"platforms" yaml:"platforms" db:"-" cli:"-"`
 }
 
+// SSHKeys returns the slide of ssh key for an application
```
Comment (bnjjj) [inline sdk/project.go]: s/slide/slice

Comment (bnjjj) [reply sdk/project.go]: s/application/project

File: sdk/project.go [hunk 10]
```diff
@@ -33,6 +33,28 @@ type Project struct {
...
+	keys := []ProjectKey{}
+	for _, k := range a.Keys {
+		if k.Type == KeyTypeSSH {
+			keys = append(keys, k)
+		}
+	}
+	return keys
+}
+
+// PGPKeys returns the slide of pgp key for an application
```
Comment (bnjjj) [inline sdk/project.go]: same

File: engine/api/bootstrap/bootstrap.go (hunk 6 above)
Comment (fsamin) [reply engine/api/bootstrap/bootstrap.go]: well i think we can't solve all the problems; we will have to check manually

File: engine/api/bootstrap/bootstrap.go (hunk 4 above)
Comment (fsamin) [reply engine/api/bootstrap/bootstrap.go]: done

File: engine/api/bootstrap/bootstrap.go (hunk 5 above)
Comment (fsamin) [reply engine/api/bootstrap/bootstrap.go]: on our CDS instance, no need to deal with them

**Pull request merged**